PyInstaller/hooks/hook-PyQt5.QtWebEngineWidgets.py

```diff
@@ -8,46 +8,58 @@
 
 import os
-from PyInstaller.utils.hooks import collect_data_files, get_qmake_path
+from PyInstaller.utils.hooks import add_qt5_dependencies, \
+    remove_prefix, get_module_file_attribute, pyqt5_library_info, \
+    collect_system_data_files
+from PyInstaller.depend.bindepend import getImports
 import PyInstaller.compat as compat
 
-hiddenimports = ["sip",
-                 "PyQt5.QtCore",
-                 "PyQt5.QtGui",
-                 "PyQt5.QtNetwork",
-                 "PyQt5.QtWebChannel",
-                 "PyQt5.QtWebEngineCore",
-                 ]
+hiddenimports, binaries, datas = add_qt5_dependencies(__file__)
 
-# Find the additional files necessary for QtWebEngine.
-datas = (collect_data_files('PyQt5', True, os.path.join('Qt', 'resources')) +
-         collect_data_files('PyQt5', True, os.path.join('Qt', 'translations')) +
-         [x for x in collect_data_files('PyQt5', False, os.path.join('Qt', 'bin'))
-          if x[0].endswith('QtWebEngineProcess.exe')])
-
-# Note that for QtWebEngineProcess to be able to find icudtl.dat the bundle_identifier
-# must be set to 'org.qt-project.Qt.QtWebEngineCore'. This can be done by passing
-# bundle_identifier='org.qt-project.Qt.QtWebEngineCore' to the BUNDLE command in
-# the .spec file. FIXME: This is not ideal and a better solution is required.
-qmake = get_qmake_path('5')
-if qmake:
-    libdir = compat.exec_command(qmake, "-query", "QT_INSTALL_LIBS").strip()
-
-    if compat.is_darwin:
-        binaries = [
-            (os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5',\
-                          'Helpers', 'QtWebEngineProcess.app', 'Contents', 'MacOS', 'QtWebEngineProcess'),
-             os.path.join('QtWebEngineProcess.app', 'Contents', 'MacOS'))
-        ]
-
-        resources_dir = os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5', 'Resources')
-        datas += [
-            (os.path.join(resources_dir, 'icudtl.dat'),''),
-            (os.path.join(resources_dir, 'qtwebengine_resources.pak'), ''),
-            # The distributed Info.plist has LSUIElement set to true, which prevents the
-            # icon from appearing in the dock.
-            (os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5',\
-                           'Helpers', 'QtWebEngineProcess.app', 'Contents', 'Info.plist'),
-                     os.path.join('QtWebEngineProcess.app', 'Contents'))
-        ]
+# Include the web engine process, translations, and resources.
+rel_data_path = 'PyQt5', 'Qt'
+if compat.is_darwin:
+    # This is based on the layout of the Mac wheel from PyPi.
+    data_path = pyqt5_library_info.location['DataPath']
+    resources = 'lib', 'QtWebEngineCore.framework', 'Resources'
+    web_engine_process = ('lib', 'QtWebEngineCore.framework', 'Helpers')
+    datas += collect_system_data_files(
+        os.path.join(data_path, *resources),
+        os.path.join(*rel_data_path, *resources[:-1]), True)
```

Comment:
1. This is valid:

```
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os.path
>>> data_path = '/home/foo/bar'
>>> resources = 'lib', 'QtWebEngineCore.framework', 'Resources'
>>> os.path.join(data_path, *resources)
'/home/foo/bar\\lib\\QtWebEngineCore.framework\\Resources'
```

2. I don't *think* I am, this is all internal Qt stuff (no Python). But I can't really say, since this is all Mac code that still doesn't fully work (sigh).